BlazarUI/app/scripts/components/sidebar/Sidebar.jsx

```diff
@@ -1,14 +1,25 @@
 import React, {Component, PropTypes} from 'react';
+import Button from 'react-bootstrap/lib/Button';
+import Icon from '../shared/Icon.jsx';
 
 class Sidebar extends Component {
 
   render() {
+    const collapseButton = (
+      <Button className="sidebar__collapse-button" onClick={this.props.collapse}>
+        <Icon for={this.props.isCollapsed ? 'right' : 'left'}></Icon>
+      </Button>
+    );
+
+    const sidebarClasses = this.props.isCollapsed ? 'sidebar-collapsed' : '';
+
     let headline;
     if (this.props.headline) {
       headline = <h2 className='sidebar__headline'>{this.props.headline}</h2>;
     }
     return (
-      <div className='sidebar'>
+      <div className={`sidebar ${sidebarClasses}`}>
+        {collapseButton}
```

Comment:
Same as above, keep `render` as declarative as possible. Adding conditional logic in here become a slippery slope. Looks like this component could make use of `renderCollapseButton` and `renderHeadline`.